figures/S3.supplemental_figure3/scripts/S3.r

```diff
@@ -0,0 +1,1060 @@
+    )
+)
+length(unique(cytokine_values_agg$oneb_Treatment_Dose_Inhibitor_Dose))
+
+length(unique(cytokine_values_agg$cytokine))
+# gsub out '[NSU]' from the cytokine names
+cytokine_values_agg$cytokine <- gsub('\\[NSU\\]', '', cytokine_values_agg$cytokine)
+length(unique(cytokine_values_agg$cytokine))
+
+
+# create a matrix of the cytokine values for the clustering and heatmap
+# unmelt the data
+cytokine_values_agg <- reshape2::dcast(cytokine_values_agg, oneb_Treatment_Dose_Inhibitor_Dose ~ cytokine, value.var = "cytokine_value", fun.aggregate = mean)
+# make oneb_Metadata_Treatment_Dose_Inhibitor_Dose the rownames
+rownames(cytokine_values_agg) <- cytokine_values_agg$oneb_Treatment_Dose_Inhibitor_Dose
+# drop the column
+cytokine_values_agg <- subset(cytokine_values_agg, select = -c(oneb_Treatment_Dose_Inhibitor_Dose))
+
+
+row_dend <- as.dendrogram(hclust(dist(cytokine_values_agg)))
+col_dend <- as.dendrogram(hclust(dist(t(cytokine_values_agg))))
+
+
+# remove '[NSU]' from the column names
+colnames(cytokine_values_agg) <- gsub("\\[NSU\\]", "", colnames(cytokine_values_agg))
+
+
+cytokine_values_agg <- as.matrix(cytokine_values_agg)
+
+
+# set plot size
+width <- 17
+height <- 40
+options(repr.plot.width=width, repr.plot.height=height)
+heatmap_plot_all <- (
+  Heatmap(
+  t(cytokine_values_agg),
+  cluster_rows = TRUE,    # Cluster rows
+  cluster_columns = TRUE, # Cluster columns
+  show_row_names = TRUE,  # Show row names
+  show_column_names = TRUE, # Show column names
+  column_names_gp = gpar(fontsize = 10), # Column name label formatting
+  row_names_gp = gpar(fontsize = 12),    # Row name label formatting
+   heatmap_legend_param = list(
+                title = "Cytokine\nAbundance", at = c(0, 1),
+                title_position = "topcenter",
+                title_gp = gpar(fontsize = 16),
+                labels_gp = gpar(fontsize = 16),
+                direction = "horizontal",
+                padding = unit(c(10, 10, 5, 5), "mm"),
+                grid_width = unit(5000, "mm"),
+                grid_height = unit(10, "mm")
+                ),
+  # make the tiles rectangular
+  rect_gp = gpar(col = NA),
+  column_dend_height = unit(4, "cm"),
+  row_dend_width = unit(4, "cm")
+    )
+)
+
+heatmap_plot_all <- draw(
+    heatmap_plot_all,
+    heatmap_legend_side = "bottom",
+    annotation_legend_side = "right",
+    # change locaiton of the legend
+    merge_legend = TRUE,
+    padding = unit(c(5, 5, 5, 15), "mm"),
+)
+
+
+png(file.path("..","figures","all_cytokines_heatmap.png"), width = width, height = height, units = "in", res = 600)
+heatmap_plot_all
+dev.off()
+heatmap_plot_all
+
+
+# load in the png to a grob object
+heatmap_grob <- png::readPNG(file.path("..","figures","all_cytokines_heatmap.png"))
+# convert the grob object to a raster grob
+# heatmap_anova_cytokines <- rasterGrob(heatmap_grob, interpolate = TRUE, height = unit(1, "npc"), width = unit(1, "npc"))
```

Comment:
Same recommendation to remove commented-out code.

Reply:
removed